plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -250,10 +241,14 @@ private Multimap<Integer, FileStatus> getCommittedPartitions(URI committedAttemp
         }
     }
 
-    private URI getExchangeDirectory(int taskPartitionId)
+    private URI getTaskOutputDirectory(int taskPartitionId)
     {
         URI baseDirectory = baseDirectories.get(taskPartitionId % baseDirectories.size());
-        return baseDirectory.resolve(exchangeContext.getQueryId() + "." + exchangeContext.getExchangeId() + PATH_SEPARATOR);
+        String randomizedPrefix = randomizedPrefixes.computeIfAbsent(taskPartitionId, ignored -> UUID.randomUUID().toString().split("-")[0]);
```

Comment:
Using `Random.nextBytes` would be more straightforward than generating long UUID and cutting it down

Reply:
`Random.nextBytes` will generate Strings that may violate S3's naming rules.